### DISCUSSION TITLE
inner variable of optional interval in other constraints

### DISCUSSION THREAD

**Discussion by gregy4** [google/or-tools#3039] [CP-SAT questions] [answered]
Hello,
  I found notable difference in duration of presolve and solve phase when start (or end) variable of optional interval was used in other constraints. Two options were considered, a constraint has to be respected only in case optional interval is active, and a constraint should be respected in every case. First option was significantly better in #2940 of fully disjunctive model.

  My initial understanding of optional interval was that nonactive optional interval should be ignored in every constraint. But it is probably not generally true when internal variable such as start (or end) variable is used.

  Is there some usecase when variables of nonactive optional interval can be used, or in every case other constraint should depend on active optional interval ?  In second case, maybe missing depency on activity of interval can be solved in presolve phase.

Thanks,
Jan

**Comment by lperron** [comment, marked as answer]
You can always use variables, whether they appear in an unperformed interval or not.

An interval variable is actually linked with a constraint that maintains:

    performed_literal implies start + size == end
